internal/resources/cluster/resource_cluster.go

```diff
@@ -295,42 +295,63 @@ func resourceClusterDelete(_ context.Context, d *schema.ResourceData, m interfac
+		newVersion := d.Get(helper.GetFirstElementOf(SpecKey, tkgVsphereClusterKey, distributionKey, versionKey))
+		cluster.Spec.TkgVsphere.Distribution.Version = newVersion.(string)
 	}
+}
 
-	if !updateRequired {
-		return diags
+func withTKGsVsphereVersionUpdate(_ authctx.TanzuContext, d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey)) {
+		newVersion := d.Get(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey))
+		cluster.Spec.TkgServiceVsphere.Distribution.Version = newVersion.(string)
 	}
+}
 
-	getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
-	if err != nil {
-		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
+func withClusterGroupUpdate(_ authctx.TanzuContext, d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, clusterGroupKey)) {
+		newClusterGroupName := d.Get(helper.GetFirstElementOf(SpecKey, clusterGroupKey))
+		if newClusterGroupName.(string) != "" {
+			cluster.Spec.ClusterGroupName = newClusterGroupName.(string)
+		}
 	}
+}
 
+func withMetaUpdate(config authctx.TanzuContext, d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
 	if common.HasMetaChanged(d) {
-		meta := common.ConstructMeta(d)
+		getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
+		if err != nil {
+			diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
+			return
+		}
+
+		objectMeta := common.ConstructMeta(d)
 
 		if value, ok := getResp.Cluster.Meta.Labels[common.CreatorLabelKey]; ok {
-			meta.Labels[common.CreatorLabelKey] = value
+			objectMeta.Labels[common.CreatorLabelKey] = value
 		}
 
-		getResp.Cluster.Meta.Labels = meta.Labels
-		getResp.Cluster.Meta.Description = meta.Description
+		cluster.Meta.Labels = objectMeta.Labels
+		cluster.Meta.Description = objectMeta.Description
+	}
+}
+
+func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	// Get call to initialise the cluster struct
+	getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
 	}
 
-	incomingCGName := d.Get(helper.GetFirstElementOf(SpecKey, clusterGroupKey))
+	var updates []func(authctx.TanzuContext, *schema.ResourceData, *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster)
```

Comment:
define this function definition and give it a name

Reply:
Done